### ISSUE TITLE
object_usage_linter doesn't match R CMD check

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#941]
As seen here: https://github.com/r-lib/lintr/pull/931#discussion_r826188056

Basically our `tools::checkUsage()` call doesn't match that of `R CMD check`. In this case, it's because the default `checkUsage()` doesn't set `skipWith=TRUE`, so objects seen inside `with()` calls can trip the linter.

My understanding is the main goal of `object_usage_linter` is to help match `R CMD check`, so we should adjust our code to match.

But maybe we think of this as a bit more general, and could expose an option instead (e.g. `match_r_check`, I would set `TRUE` by default).

**Comment by MichaelChirico** [comment]
PS I think this linter should be tagged as `package_development`

**Comment by AshesITR** [comment]
Is there a legitimate true positive caused by `skipWith = FALSE`?
I reckon most `with()` calls are using columns from a data frame and thus always trip `checkUsage()` unless the used columns are declared outside of the data frame as well.

BTW:

```r
checkUsage(fun, name = "<anonymous>", report = cat, all = FALSE, 
           suppressLocal = FALSE, suppressParamAssigns = !all,
           suppressParamUnused = !all, suppressFundefMismatch = FALSE,
           suppressLocalUnused = FALSE, suppressNoLocalFun = !all,
           skipWith = FALSE, suppressUndefined = dfltSuppressUndefined,
           suppressPartialMatchArgs = TRUE) 
```

`suppressPartialMatchArgs = FALSE` sounds interesting.
Also,  `suppressLocalUnused = symbols_in_glue_calls` seems like the way to support `glue::glue()`.
